packages/compiler/src/render3/view/styling_builder.ts

```diff
@@ -219,7 +219,11 @@ export class StylingBuilder {
     if (isEmptyExpression(value)) {
       return null;
     }
-    name = normalizePropName(name);
+    // CSS custom properties are case-sensitive so we shouldn't normalize them.
+    // See: https://www.w3.org/TR/css-variables-1/#defining-variables
+    if (!isCssCustomProperty(name)) {
+      name = normalizePropName(name);
```

Comment:
nit: it looks like `normalizePropName` is just a wrapper around `hyphenate` function, so I'd propose dropping it and use `hyphenate` directly:
```suggestion
      name = hyphenate(name);
```